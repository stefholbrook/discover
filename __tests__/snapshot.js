import React from 'react'
import renderer from 'react-test-renderer'
import Index from '../pages/index'
import localArtists from '../__mocks__/artists.json'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')
const useQuery = jest.spyOn(require('@apollo/client'), 'useQuery')

it('renders homepage unchanged', () => {
  useRouter.mockImplementationOnce(() => ({
    query: { search: 'eyJsb2NhdGlvbiI6InNhbHQgbGFrZSBjaXR5IiwiZGVjYWRlIjoiMjAxNyBUTyAyMDIyIn0=' },
  }))
  useQuery.mockImplementationOnce(() => ({ localArtists }))

  const tree = renderer.create(<Index />).toJSON()
  expect(tree).toMatchSnapshot()
})
