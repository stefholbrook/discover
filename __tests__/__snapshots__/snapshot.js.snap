// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
Array [
  <div
    className="pages__StyledHero-sc-1cvwks4-0 drnWYN"
  >
    <nav
      className="pages__StyledHeader-sc-1cvwks4-6 fRalg"
    >
      <img
        className="pages__StyledLogo-sc-1cvwks4-7 irkHFk"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        src="/images/daemo_logo.png"
      />
      <div
        className="pages__StyledRightMenu-sc-1cvwks4-9 tLDZh"
      >
        <a
          className="pages__StyledHeaderLink-sc-1cvwks4-8 cANlmA"
          href="/about"
        >
          About
        </a>
      </div>
    </nav>
    <div
      className="pages__StyledSiteHeader-sc-1cvwks4-1 jYMAAc"
    >
      <div
        className="pages__StyledIntroContent-sc-1cvwks4-2 brNEUe"
      >
        <p
          className="pages__IntroDesc-sc-1cvwks4-3 enBrvL"
        >
          daemo.
        </p>
        <h1
          className="pages__StyledIntroTitle-sc-1cvwks4-4 cUhJWT"
        >
          Discover | The best music you've never heard.
        </h1>
        <form
          onSubmit={[Function]}
        >
          <label
            className="query-form__StyledSelectLabel-sc-18cy6nr-3 hpTFFv"
            htmlFor="decade"
          />
          <select
            className="query-form__StyledSelect-sc-18cy6nr-2 grqgID"
            id="decade"
            onChange={[Function]}
            placeholder="decade"
            value=""
          >
            <option
              disabled={true}
              hidden={true}
              value=""
            >
              begin date
            </option>
            <option
              value="2020 TO 2021"
            >
              Last year
            </option>
            <option
              value="2017 TO 2022"
            >
              Last 5 years
            </option>
            <option
              value="2010 TO 2019"
            >
              2010s
            </option>
            <option
              value="2000 TO 2009"
            >
              2000s
            </option>
            <option
              value="1990 TO 1999"
            >
              1990s
            </option>
            <option
              value="1980 TO 1989"
            >
              1980s
            </option>
            <option
              value="1970 TO 1979"
            >
              1970s
            </option>
          </select>
          <label
            className="query-form__StyledSelectLabel-sc-18cy6nr-3 hpTFFv"
            htmlFor="search"
          />
          <input
            className="query-form__StyledInput-sc-18cy6nr-1 hobEaH"
            id="search"
            onChange={[Function]}
            placeholder="search city, state, or country"
            size="30"
            type="search"
            value=""
          />
          <button
            className="query-form__StyledButton-sc-18cy6nr-4 jDCqYz"
            disabled={false}
            type="submit"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-search fa-w-16 "
              data-icon="search"
              data-prefix="fas"
              focusable="false"
              onClick={[Function]}
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </form>
      </div>
    </div>
  </div>,
  <div
    className="pages__StyledContainer-sc-1cvwks4-5 jzMjqw"
  >
    <div>
      <div
        className="feed__StyledSectionHeader-sc-1p7trc5-1 dHBFGw"
      >
        <h2
          className="feed__StyledTitle-sc-1p7trc5-2 bglvwd"
        >
          Results
        </h2>
        <h2
          className="feed__StyledSubtitle-sc-1p7trc5-3 dzoSsJ"
        >
          Filters: 
          salt lake city
          , 
          2017 TO 2022
        </h2>
      </div>
      <div
        className="feed__StyledArtistContainer-sc-1p7trc5-0 kjYxNu"
        onScroll={[Function]}
      />
    </div>
  </div>,
]
`;
